Basic-Car-Maintenance/Shared/Managers/AnayticsManager.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  AnayticsManager.swift
+//  Basic-Car-Maintenance
+//
+//  Created by 0xJs on 10/11/23.
+//
+import Foundation
+import FirebaseAnalytics
+
+/// Manager to handle app Analytics
+final class AnalyticsManager {
+    private init() {}
+    
+    static let shared = AnalyticsManager()
+    
+    public func logEvent(_ event: AnalyticsAction, parameters: [String: Any] = [:]) {
```

Comment:
This doesn't need to be `public`. and for logging an event, I don't think we need to include the screen, because that may change sometime.
Plus there's already a different issue for analytics for each specific screen